nucypher/blockchain/eth/deployers.py

```diff
@@ -293,90 +292,224 @@ class PolicyManagerDeployer(ContractDeployer):
+        wrapped_policy_manager_contract = self.blockchain.interface. \
+            _wrap_contract(proxy_contract, target_contract=policy_manager_contract)
 
         # Switch the contract for the wrapped one
-        the_policy_manager_contract = wrapped_policy_manager_contract
+        policy_manager_contract = wrapped_policy_manager_contract
 
         # Configure the MinerEscrow by setting the PolicyManager
-        policy_setter_txhash = self.miner_agent.contract.functions. \
-            setPolicyManager(the_policy_manager_contract.address).transact({'from': self.deployer_address})
+        policy_setter_txhash = self.miner_agent.contract.functions.setPolicyManager(policy_manager_contract.address) \
+            .transact({'from': self.deployer_address})
 
         self.blockchain.wait_for_receipt(policy_setter_txhash)
 
         # Gather the transaction hashes
         deployment_transactions = {'deployment': deploy_txhash,
-                                   'dispatcher_deployment': dispatcher_deploy_txhash,
+                                   'dispatcher_deployment': proxy_deploy_txhashes['txhash'],
                                    'set_policy_manager': policy_setter_txhash}
 
         self.deployment_transactions = deployment_transactions
-        self._contract = the_policy_manager_contract
+        self._contract = policy_manager_contract
 
         return deployment_transactions
 
 
-class UserEscrowDeployer(ContractDeployer):
-    """
-    TODO: Consider Agency amongst many user escrows,
-    goverment, token transfer, and deployment
+class LibraryLinkerDeployer(ContractDeployer):
 
-    Depends on Token, MinerEscrow, and PolicyManager
-    """
+    _contract_name = 'UserEscrowLibraryLinker'
+    __is_proxy = True
+    __upgradeable = False
+
+    def __init__(self, target_contract: Contract, secret_hash: bytes, *args, **kwargs):
+        self.target_contract = target_contract
+        self.secret_hash = secret_hash
+        super().__init__(*args, **kwargs)
+
+    def deploy(self) -> dict:
+        linker_args = (self._contract_name, self.target_contract.address, self.secret_hash)
+        linker_contract, linker_deployment_txhash = self.blockchain.interface.deploy_contract(*linker_args)
+        self._contract = linker_contract
+        return {'txhash': linker_deployment_txhash}
+
+
+class UserEscrowProxyDeployer(ContractDeployer):
+
+    _contract_name = 'UserEscrowProxy'
+    __is_proxy = True
+    __upgradeable = True
+    __proxy_deployer = LibraryLinkerDeployer
+
+    def __init__(self, secret_hash: bytes, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.token_agent = NucypherTokenAgent(blockchain=self.blockchain)
+        self.miner_agent = MinerAgent(blockchain=self.blockchain)
+        self.policy_agent = PolicyAgent(blockchain=self.blockchain)
+        self.secret_hash = secret_hash
+
+    def __get_state_contract(self) -> str:
+        return self.contract.functions.getStateContract()
```

Comment:
What purpose of this method?  it's strange for me to see line without `call` or `transact`